frame/contracts/src/exec.rs

```diff
@@ -831,14 +834,19 @@ where
 		let do_transaction = || {
 			// We need to charge the storage deposit before the initial transfer so that
 			// it can create the account in case the initial transfer is < ed.
-			if entry_point == ExportedFunction::Constructor {
-				let frame = top_frame_mut!(self);
-				frame.nested_storage.charge_instantiate(
-					&self.origin,
-					&frame.account_id,
-					frame.contract_info.get(&frame.account_id),
-				)?;
-			}
+			// Also, we can only charge the storage deposit if there is an account id associated
+			// with the origin, if the origin is Root there is no storage deposit to charge.
+			match (entry_point, &self.origin) {
+				(ExportedFunction::Constructor, Origin::Signed(origin)) => {
```

Comment:
Nit: Rewriting this as `if let (ExportedFunction::Constructor, Origin::Signed(origin)) = (entry_point, &self.origin)` would remove one nesting layer

Reply:
This bit has been refactored